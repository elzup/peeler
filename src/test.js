import m from '.'

test('works', () => {
  expect(m('before(hit)after')).toMatchSnapshot('single')
  expect(m('aa(bb(cc(dd)cc)bb)aa')).toMatchSnapshot('three nest')
})

test('complex', () => {
  expect(m('')).toMatchSnapshot('empty')
  expect(m('()')).toMatchSnapshot('sub empty')
  expect(m('(())')).toMatchSnapshot('sub sub empty')
  expect(m('({}fuga{})')).toMatchSnapshot('nest side')
  expect(m('aa({cc}bb{c2(dd)c2}b)aa(b2)aa')).toMatchSnapshot('gata gata')
  expect(m('[({})]')).toMatchSnapshot('any bracket')
})

test('escape works', () => {
  console.log('\\'.length)
  expect(m('\\')).toMatchSnapshot()
  expect(m('(\\(OK\\)\\[GOOD\\])')).toMatchSnapshot()
})

test('bracket options', () => {
  expect(
    m('aaa(bbb{})ccc[ddd]', {
      pairs: ['()'],
    })
  ).toMatchSnapshot('() only')
  expect(
    m('tOshinO kyoko', {
      pairs: ['Oo'],
    })
  ).toMatchSnapshot('rare option')
})

test('nestMax options', () => {
  expect(
    m('((()))', {
      nestMax: 2,
    })
  ).toMatchSnapshot('stop nest 2')
  expect(
    m('((()))', {
      nestMax: 1,
    })
  ).toMatchSnapshot('stop nest 1')
})

test('error', () => {
  // '('
  // 'hog{e(b}c)d'
})
