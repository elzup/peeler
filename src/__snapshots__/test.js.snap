// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bracket options: () only 1`] = `
Array [
  Object {
    "content": "aaa",
    "depth": 0,
    "end": 3,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 9,
    "nodes": Array [
      Object {
        "content": "bbb{}",
        "depth": 1,
        "end": 9,
        "start": 4,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 3,
    "type": "bracket",
  },
  Object {
    "content": "ccc[ddd]",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`complex: any bracket 1`] = `
Array [
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": "]",
    "depth": 0,
    "end": 5,
    "nodes": Array [
      Object {
        "content": "",
        "depth": 1,
        "end": 1,
        "start": 1,
        "type": "text",
      },
      Object {
        "close": ")",
        "depth": 1,
        "end": 4,
        "nodes": Array [
          Object {
            "content": "",
            "depth": 2,
            "end": 2,
            "start": 2,
            "type": "text",
          },
          Object {
            "close": "}",
            "depth": 2,
            "end": 3,
            "nodes": Array [
              Object {
                "content": "",
                "depth": 3,
                "end": 3,
                "start": 3,
                "type": "text",
              },
            ],
            "open": "{",
            "start": 2,
            "type": "bracket",
          },
          Object {
            "content": "",
            "depth": 2,
            "end": 4,
            "start": 4,
            "type": "text",
          },
        ],
        "open": "(",
        "start": 1,
        "type": "bracket",
      },
      Object {
        "content": "",
        "depth": 1,
        "end": 5,
        "start": 5,
        "type": "text",
      },
    ],
    "open": "[",
    "start": 0,
    "type": "bracket",
  },
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`complex: empty 1`] = `
Array [
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`complex: gata gata 1`] = `
Array [
  Object {
    "content": "aa",
    "depth": 0,
    "end": 2,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 20,
    "nodes": Array [
      Object {
        "content": "",
        "depth": 1,
        "end": 3,
        "start": 3,
        "type": "text",
      },
      Object {
        "close": "}",
        "depth": 1,
        "end": 6,
        "nodes": Array [
          Object {
            "content": "cc",
            "depth": 2,
            "end": 6,
            "start": 4,
            "type": "text",
          },
        ],
        "open": "{",
        "start": 3,
        "type": "bracket",
      },
      Object {
        "content": "bb",
        "depth": 1,
        "end": 9,
        "start": 7,
        "type": "text",
      },
      Object {
        "close": "}",
        "depth": 1,
        "end": 18,
        "nodes": Array [
          Object {
            "content": "c2",
            "depth": 2,
            "end": 12,
            "start": 10,
            "type": "text",
          },
          Object {
            "close": ")",
            "depth": 2,
            "end": 15,
            "nodes": Array [
              Object {
                "content": "dd",
                "depth": 3,
                "end": 15,
                "start": 13,
                "type": "text",
              },
            ],
            "open": "(",
            "start": 12,
            "type": "bracket",
          },
          Object {
            "content": "c2",
            "depth": 2,
            "end": 18,
            "start": 16,
            "type": "text",
          },
        ],
        "open": "{",
        "start": 9,
        "type": "bracket",
      },
      Object {
        "content": "b",
        "depth": 1,
        "end": 20,
        "start": 19,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 2,
    "type": "bracket",
  },
  Object {
    "content": "aa",
    "depth": 0,
    "end": 23,
    "start": 21,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 26,
    "nodes": Array [
      Object {
        "content": "b2",
        "depth": 1,
        "end": 26,
        "start": 24,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 23,
    "type": "bracket",
  },
  Object {
    "content": "aa",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`complex: nest side 1`] = `
Array [
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 9,
    "nodes": Array [
      Object {
        "content": "",
        "depth": 1,
        "end": 1,
        "start": 1,
        "type": "text",
      },
      Object {
        "close": "}",
        "depth": 1,
        "end": 2,
        "nodes": Array [
          Object {
            "content": "",
            "depth": 2,
            "end": 2,
            "start": 2,
            "type": "text",
          },
        ],
        "open": "{",
        "start": 1,
        "type": "bracket",
      },
      Object {
        "content": "fuga",
        "depth": 1,
        "end": 7,
        "start": 3,
        "type": "text",
      },
      Object {
        "close": "}",
        "depth": 1,
        "end": 8,
        "nodes": Array [
          Object {
            "content": "",
            "depth": 2,
            "end": 8,
            "start": 8,
            "type": "text",
          },
        ],
        "open": "{",
        "start": 7,
        "type": "bracket",
      },
      Object {
        "content": "",
        "depth": 1,
        "end": 9,
        "start": 9,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 0,
    "type": "bracket",
  },
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`complex: sub empty 1`] = `
Array [
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 1,
    "nodes": Array [
      Object {
        "content": "",
        "depth": 1,
        "end": 1,
        "start": 1,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 0,
    "type": "bracket",
  },
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`complex: sub sub empty 1`] = `
Array [
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 3,
    "nodes": Array [
      Object {
        "content": "",
        "depth": 1,
        "end": 1,
        "start": 1,
        "type": "text",
      },
      Object {
        "close": ")",
        "depth": 1,
        "end": 2,
        "nodes": Array [
          Object {
            "content": "",
            "depth": 2,
            "end": 2,
            "start": 2,
            "type": "text",
          },
        ],
        "open": "(",
        "start": 1,
        "type": "bracket",
      },
      Object {
        "content": "",
        "depth": 1,
        "end": 3,
        "start": 3,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 0,
    "type": "bracket",
  },
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`nestMax options: stop nest 1 1`] = `
Array [
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 3,
    "nodes": Array [
      Object {
        "content": "((",
        "depth": 1,
        "end": 3,
        "start": 1,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 0,
    "type": "bracket",
  },
  Object {
    "content": "))",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`nestMax options: stop nest 2 1`] = `
Array [
  Object {
    "content": "",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 4,
    "nodes": Array [
      Object {
        "content": "",
        "depth": 1,
        "end": 1,
        "start": 1,
        "type": "text",
      },
      Object {
        "close": ")",
        "depth": 1,
        "end": 3,
        "nodes": Array [
          Object {
            "content": "(",
            "depth": 2,
            "end": 3,
            "start": 2,
            "type": "text",
          },
        ],
        "open": "(",
        "start": 1,
        "type": "bracket",
      },
      Object {
        "content": "",
        "depth": 1,
        "end": 4,
        "start": 4,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 0,
    "type": "bracket",
  },
  Object {
    "content": ")",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`works: single 1`] = `
Array [
  Object {
    "content": "before",
    "depth": 0,
    "end": 6,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 10,
    "nodes": Array [
      Object {
        "content": "hit",
        "depth": 1,
        "end": 10,
        "start": 7,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 6,
    "type": "bracket",
  },
  Object {
    "content": "after",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;

exports[`works: three nest 1`] = `
Array [
  Object {
    "content": "aa",
    "depth": 0,
    "end": 2,
    "start": 0,
    "type": "text",
  },
  Object {
    "close": ")",
    "depth": 0,
    "end": 17,
    "nodes": Array [
      Object {
        "content": "bb",
        "depth": 1,
        "end": 5,
        "start": 3,
        "type": "text",
      },
      Object {
        "close": ")",
        "depth": 1,
        "end": 14,
        "nodes": Array [
          Object {
            "content": "cc",
            "depth": 2,
            "end": 8,
            "start": 6,
            "type": "text",
          },
          Object {
            "close": ")",
            "depth": 2,
            "end": 11,
            "nodes": Array [
              Object {
                "content": "dd",
                "depth": 3,
                "end": 11,
                "start": 9,
                "type": "text",
              },
            ],
            "open": "(",
            "start": 8,
            "type": "bracket",
          },
          Object {
            "content": "cc",
            "depth": 2,
            "end": 14,
            "start": 12,
            "type": "text",
          },
        ],
        "open": "(",
        "start": 5,
        "type": "bracket",
      },
      Object {
        "content": "bb",
        "depth": 1,
        "end": 17,
        "start": 15,
        "type": "text",
      },
    ],
    "open": "(",
    "start": 2,
    "type": "bracket",
  },
  Object {
    "content": "aa",
    "depth": 0,
    "end": 0,
    "start": 0,
    "type": "text",
  },
]
`;
